tests/lib/processor.js

```diff
@@ -362,7 +362,7 @@ describe("processor", () => {
             const blocks = processor.preprocess(code);
 
             assert.strictEqual(blocks[0].filename, "0.js");
-            assert.strictEqual(blocks[0].text, "var answer = 6 * 7;\nconsole.log(answer);\n");
+            assert.strictEqual(blocks[0].text, "var answer = 6 * 7;\r\nconsole.log(answer);\n");
```

Comment:
This test is... strange. It says it "preserves" the original line endings, yet the assertion _normalized_ them to `\n`. However, it was consistent. Here I see there's one `\r\n` and another `\n`. Any idea why that is? If v9 of the parser now preserves original line endings, I'd expect both to be `\r\n`. If this behavior changes, I'll want to add some autofix tests that span lines to make sure we're using correct substitutions. That'd be a likely place to have an off-by-one error that causes syntax errors in the output.

Reply:
Yep, I agree it's weird. I think the last line of the code block is always normalized to `\n`, while the other line endings are preserved. It can be observed by adding another line inside the code block. Looks like an upstream issue.

Reply:
Oh no, it's actually due to this:

https://github.com/eslint/eslint-plugin-markdown/blob/f6a8ca0824b8a62d0af2a9eb512104564588b49e/lib/processor.js#L256-L260

It always adds a blank line and to the node value, and use `\n` without checking if it's relevant. I'm not sure why we add a blank line here…

For reference, here's the snippet I used to check the upstream implementation is working as expected:

```js
console.log(
  require('mdast-util-from-markdown')([
    "```js",
    "var answer = 6 * 7;",
    "console.log(answer);",
    "```"
  ].join("\r\n")).children[0].value
); // 'var answer = 6 * 7;\r\nconsole.log(answer);'
```